### DISCUSSION TITLE
plugin-tiled Loading .tmx file

### DISCUSSION THREAD

**Discussion by densetsuuu** [excaliburjs/Excalibur#2843] [Q&A]
Hello, i'm trying to develop a 2D roguelike game with ExcaliburJS, i want to import a .tmx (tiled map) file inside my game. 
I'm using vite and pnpm. 
Excalibur version: 0.28.2
plugin-tiled: 0.28.0

For that, i use this function:
```typescript
public startCustomLoader() {
		console.log(this.map);
		const loader: Loader = new Loader([this.map]);
		this.logLoadingProgress(loader);
		return Game.getInstance().start(loader);
}
```
I have no path problems when importing the .tmx tiled map file, and when printing the map object before creating the loader, i get this:
https://github.com/excaliburjs/Excalibur/assets/103574286/89796579-b633-466a-8d81-1b780d161c3f
So, i assume that the TiledMapResource object is well created.

The problem is that the Loader is frozen at 0%, and the logLoadingProgress function is running forever, 
![image](https://github.com/excaliburjs/Excalibur/assets/103574286/f435997a-b6ef-46f4-b21c-4824db5f4f73)

Could you help me figure out what's the issue here ?

PS: I'm new in the game development world

Thanks by advance !

**Comment by densetsuuu** [comment]
Solved, i used this repo: https://github.com/excaliburjs/sample-tiled-vite/tree/main?tab=readme-ov-file that helped me very much !

**Reply by eonarheim** [reply]
@raphael-alarcon Glad that helped! What was the issue by the way? Perhaps we can update the docs to make this easier

**Reply by densetsuuu** [reply]
@eonarheim The issue was the .tmx file ! I used the example-city.tmx file that is used in the documentation: https://excaliburjs.com/docs/tilemap/

**Reply by densetsuuu** [reply]
It appears that in the case where the .tmx file is invalid (like if the specified tileset is invalid for example), the loader just freezes because it cannot read this file. One thing that could be interesting would be to say that the loader has troubles loading the file in the console.
